firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -493,6 +493,22 @@ class BrowserViewController: UIViewController,
         scrollController.showToolbars(animated: true)
     }
 
+    @objc
+    func sceneDidEnterBackgroundNotification(notification: Notification) {
+        // Ensure the notification is for the current window scene
+        guard let currentWindowScene = view.window?.windowScene,
+              let notificationWindowScene = notification.object as? UIWindowScene,
+              currentWindowScene === notificationWindowScene else { return }
```

Comment:
Nit: I think it would probably be better just to compare the `uuid` of the BrowserWindow to the UUID of the scene rather than do this pointer comparison of the scene objects. I would need to refresh my memory though on how that is all wired up, doesn't need to block this PR.